### PR TITLE
detect unneeded retests

### PR DESCRIPTION
**Pull request by lavalamp**
tested! works!

### CODE REVIEW

**Comment by lavalamp** [comment]
@eparis ready for a review!

**Comment by lavalamp** [comment]
I don't have much patience-- I built & pushed this just now... I promise to fix anything @eparis wants changed.

**Comment by eparis** [comment]
If its already live we should just merge it, so we can at least find the git commit! (I totally understand, I want to do the same ALL THE TIME)

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -678,6 +678,39 @@ func (obj *MungeObject) RemoveLabel(label string) error {
 	return nil
 }
 
+// Returns the head SHA and the base ref, so that you can get the base's sha in
```
Comment (eparis) [inline mungegithub/github/github.go]: golint requires this comment to start with `// GetHeadAndBase .....`

**Comment by lavalamp** [comment]
Fixed.

**Comment by eparis** [comment]
let me write something to show 'RetestsAvoided' in the webUI....

**Comment by lavalamp** [comment]
Sure, or we can wait to see if it actually ever avoids any...

**Comment by lavalamp** [comment]
This got hit for the first time live! Every time this happens we've saved ~30 minutes.

screenshot 2016-05-17 at 10 33 19 pm

**Pull request merged**